### PR TITLE
Chain state initialisation

### PR DESCRIPTION
**Pull request by AriFordsham**

### CODE REVIEW

**Comment by AriFordsham** [comment]
The monad code could do with a bit of tidying - it might need a monad transformer somewhere.

@Geometer1729 let me know what you think.

File: examples/Spec.hs [hunk 1]
```diff
@@ -18,37 +15,46 @@ main =
...
+                )
+          )
+          (Model.initMock Model.defaultAlonzo (lovelaceValue 101 <> nft))
+  txTestGoodAdjunction auctionTest init _ _
+
+lovelaceValue :: Integer -> Plutus.Value
+lovelaceValue = Plutus.singleton Plutus.adaSymbol Plutus.adaToken
+
+nft :: Plutus.Value
+nft = Plutus.singleton (Plutus.CurrencySymbol "NFT") "" 1
```
Comment (Geometer1729) [inline examples/Spec.hs]: The `IsString` instance for `BuiltinBytestring` attempts to decode a hex string so this probably needs to be a valid hex string and might need to be the correct length for a currency symbol (28 bytes or 56 hex characters) to not throw any weird errors.

**Review by Geometer1729 — CHANGES_REQUESTED** [review]
Other than the `(Plutus.CurrencySymbol "NFT")` LGTM

**Comment by Geometer1729** [comment]
I'm afraid the token name is also a hex string.

**Comment by AriFordsham** [comment]
It has a custom `IsString` instance that encodes it correctly - http://localhost:8080/plutus-ledger-api/html/src/PlutusLedgerApi.V1.Value.html#TokenName:~:text=Doesn%27t%20verify%20length.-,instance,-IsString%20TokenName%20where

Currency symbols are a hash, so are pseudo-random bytestrings, whilst token names are dev-defined and intended to be human-readable.

**Comment by Geometer1729** [comment]
> It has a custom `IsString` instance that encodes it correctly - http://localhost:8080/plutus-ledger-api/html/src/PlutusLedgerApi.V1.Value.html#TokenName:~:text=Doesn%27t%20verify%20length.-,instance,-IsString%20TokenName%20where
> 
> Currency symbols are a hash, so are pseudo-random bytestrings, whilst token names are dev-defined and intended to be human-readable.

Oh okay, sorry about that.

**Review by Geometer1729 — APPROVED** [review]
LGTM

**Pull request merged**